### ISSUE TITLE
Implement short-circuit evaluation for AND expression

### ISSUE THREAD

**Issue by Masterlvng** [clarkmcc/cel-rust#117]
Description:

In the provided example, the AND expression within the compiled program causes an error, which is counterintuitive and violates common programming language conventions.
```rust
let mut context = Context::default();
let data: HashMap<String, String>  = HashMap::new();
context.add_variable_from_value("data", data);
let program = Program::compile("has(data.x) && data.x.startsWith(\"foo\")").unwrap();
```

Expected Behavior:

The AND expression should support short-circuit evaluation. In this case, the data.x.startsWith("foo") part should only be evaluated if has(data.x) returns true. This would prevent unnecessary errors and align with typical programming language behavior.

I have prepared a pull request to address this issue by implementing short-circuit evaluation for AND expressions.

Thank you for your time and effort in maintaining this project.

**Comment by clarkmcc** [comment]
Great idea, will review tomorrow!

**Comment by Masterlvng** [comment]
> Great idea, will review tomorrow!
Hi @clarkmcc  
If there's anything I can do to help or if you need more information, please let me know.
Wish you a merry Christmas

**Comment by clarkmcc** [comment]
Sorry! Christmas traveling distracted me. I left two comments just based on an initial reading from my phone, haven't had a chance to really dive in yet. Also it looks like the linter is failing so if you could resolve that, that would be great.

**Comment by Masterlvng** [comment]
I'm sorry for interrupting your Christmas trip. Have a wonderful holiday and feel free to review again when you are available.
